adc/ade7913/ade7913-test.c

```diff
@@ -14,64 +14,108 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <string.h>
 
 #include <sys/msg.h>
+#include <imxrt-multi.h>
 
 #include "ade7913.h"
 
 
+#define ADE7913_DEFAULT_LIST "1230"
+
+
+static int lpspi_config(oid_t *device, int spi)
+{
+	msg_t msg;
+	multi_i_t *imsg = (multi_i_t *)msg.i.raw;
+
+	msg.type = mtDevCtl;
+	msg.i.data = NULL;
+	msg.i.size = 0;
+	msg.o.data = NULL;
+	msg.o.size = 0;
+
+	imsg->id = id_spi1 + spi;
+	imsg->spi.type = spi_config;
+	imsg->spi.config.cs = 0;
+	imsg->spi.config.mode = spi_mode_0;
+	imsg->spi.config.endian = spi_msb;
+	imsg->spi.config.sckDiv = 4;
+	imsg->spi.config.prescaler = 3;
+
+	return msgSend(device->port, &msg);
+}
+
+
 int main(int argc, char **argv)
 {
-	int i;
+	int i, devnum, devcnt;
+	const char *order;
 	oid_t ade7913_spi;
 	ade7913_burst_reg_t sample;
 
-	int devcnt = 1;
-
-	if (argc >= 2)
-		devcnt = atoi(argv[1]);
-
-	printf("Starting ADE7913 test (%d devices)\n", devcnt);
+	if (argc != 2) {
+		printf("No device list given. Using default: %s\n", ADE7913_DEFAULT_LIST);
+		devcnt = strlen(ADE7913_DEFAULT_LIST);
+		order = ADE7913_DEFAULT_LIST;
+	} else {
+		printf("Device order: %s\n", ADE7913_DEFAULT_LIST);
+		devcnt = strlen(argv[1]);
+		order = argv[1];
+	}
 
-	if (devcnt > 4 || devcnt <= 0) {
+	if (devcnt > 4) {
 		printf("Incorrect ADE7913 device count (4 max)\n");
 		return -1;
 	}
 
 	while (lookup("/dev/spi1", NULL, &ade7913_spi) < 0)
 		usleep(5000);
 
+	if (lpspi_config(&ade7913_spi, 0) < 0) {
+		printf("Could not initialize SPI1\n");
+		return -1;
+	}
+
 	printf("SPI1 initialized\n");
 
+	/* Start init from device with xtal */
 	for (i = 0; i < devcnt; ++i) {
-		printf("Configuring ADE7913 device nr %d\n", i);
+		devnum = (int)(order[i] - '0');
```

Comment:
You should check if you got` 0 >= devnum > devcnt`. If I were to pass "abcd" as dev list, this wouldn't end well

Reply:
Added check for order (from '0' to '3').